Digizuite/Metadata/MetaFieldService.cs

```diff
@@ -23,6 +26,61 @@ public class MetaFieldService : IMetaFieldService
+                MultiComboValueMetaFieldResponse  => new MultiComboValueMetaFieldResponse(),
+                ComboValueMetaFieldResponse comboValueMetaFieldResponse => new ComboValueMetaFieldResponse { ViewType = comboValueMetaFieldResponse.ViewType},
+                MasterItemReferenceMetaFieldResponse masterItemReferenceMetaFieldResponse =>
+                    new MasterItemReferenceMetaFieldResponse 
+                    { 
+                        ItemType = masterItemReferenceMetaFieldResponse.ItemType,
+                        MaxCount = masterItemReferenceMetaFieldResponse.MaxCount,
+                        RelatedMetaFieldLabelId = masterItemReferenceMetaFieldResponse.RelatedMetaFieldLabelId
+                    },
+                SlaveItemReferenceMetaFieldResponse slaveItemReferenceMetaFieldResponse => 
+                    new SlaveItemReferenceMetaFieldResponse
+                    {
+                        ItemType = slaveItemReferenceMetaFieldResponse.ItemType,
+                        RelatedMetaFieldLabelId = slaveItemReferenceMetaFieldResponse.RelatedMetaFieldLabelId
+                    },
+                FloatMetaFieldResponse => new FloatMetaFieldResponse(),
+                LinkMetaFieldResponse => new LinkMetaFieldResponse(),
+                TreeMetaFieldResponse treeMetaFieldResponse => new TreeMetaFieldResponse { RecursiveToRoot = treeMetaFieldResponse.RecursiveToRoot },
+                EditMultiComboValueMetaFieldResponse => new EditMultiComboValueMetaFieldResponse(),
+                EditComboValueMetaFieldResponse => new EditComboValueMetaFieldResponse(),
+                _ => throw new ArgumentOutOfRangeException(nameof(metaFieldResponse),
+                        $"Unknown metafield value type. Got {metaFieldResponse.GetType()}")
+            };
+
+            field.MetafieldId = metaFieldResponse.MetafieldId;
+            field.SortIndex = metaFieldResponse.SortIndex;
+            field.Required = metaFieldResponse.Required;
+            field.Readonly = metaFieldResponse.Readonly;
+            field.AutoTranslated = metaFieldResponse.AutoTranslated;
+            field.AutoTranslatedOverwriteExisting = metaFieldResponse.AutoTranslatedOverwriteExisting;
+            field.VisibilityMetaFieldId = metaFieldResponse.VisibilityMetaFieldId;
+            field.VisibilityRegex = metaFieldResponse.VisibilityRegex;
+            field.GroupId = metaFieldResponse.GroupId;
+            field.Guid = metaFieldResponse.Guid;
+            field.ItemId = metaFieldResponse.ItemId;
+            field.RestrictToItemType = metaFieldResponse.RestrictToItemType;
+            field.Audited = metaFieldResponse.Audited;
+#pragma warning disable 612
+            field.LabelId = label.LabelId;
+            field.LanguageId = label.LanguageId;
+            field.Label = label.Label;
+#pragma warning restore 612
+
+            return field;
+        }
```

Comment:
I think all of this can be replaced with this, since these are records:
```suggestion
            return metaFieldResponse with
            {
#pragma warning disable 612
                LabelId = label.LabelId,
                LanguageId = label.LanguageId,
                Label = label.Label,
#pragma warning restore 612
            };
        }
```
Or at least something like it. Records are smart enough to keep the correct subtype when using `with`.